java/rocksjni/backupablejni.cc

```diff
@@ -105,10 +105,9 @@ jintArray Java_org_rocksdb_BackupableDB_getCorruptedBackups(
   // store backupids in int array
   const std::vector<rocksdb::BackupID>::size_type
       kIdSize = backup_ids.size();
-  int int_backup_ids[kIdSize];
-  for (std::vector<rocksdb::BackupID>::size_type i = 0;
-      i != kIdSize; i++) {
-    int_backup_ids[i] = backup_ids[i];
+  std::auto_ptr<jint> int_backup_ids(new jint[kIdSize]);
```

Comment:
We don't use auto_ptr in our code base. Can you explain why you don't simply use std::vector<jint>?